pallets/passkey/src/lib.rs

```diff
@@ -131,28 +146,40 @@ pub mod module {
 	#[pallet::validate_unsigned]
 	impl<T: Config> ValidateUnsigned for Pallet<T>
 	where
-		<T as frame_system::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+		<T as frame_system::Config>::RuntimeCall:
+			From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
 	{
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
 			let payload = Self::filter_valid_calls(&call)?;
 			Self::validate_signatures(&payload)?;
-
 			let nonce_check = PasskeyNonce::new(payload.passkey_call.clone());
-			nonce_check.validate()
+			nonce_check.validate()?;
+			let tx_charge = ChargeTransactionPayment::<T>(
+				payload.passkey_call.account_id.clone(),
+				call.clone(),
+			);
+			tx_charge.validate()
 		}
 
 		fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
-			Self::validate_unsigned(TransactionSource::InBlock, call)?;
-
 			let payload = Self::filter_valid_calls(&call)?;
+			Self::validate_signatures(&payload)?;
 			let nonce_check = PasskeyNonce::new(payload.passkey_call.clone());
-			nonce_check.pre_dispatch()
+			nonce_check.pre_dispatch()?;
+			let tx_charge = ChargeTransactionPayment::<T>(
+				payload.passkey_call.account_id.clone(),
+				call.clone(),
+			);
+			tx_charge.pre_dispatch()
```

Comment:
Same as the validate, I think we would need to check the signed extensions to see how they result the results of these.

Reply:
i think it wont matter in pre_dispatch since it returns an empty result in pre_dispatch